arithmetization/src/test/java/net/consensys/linea/zktracer/instructionprocessing/callTests/sixtyThreeSixtyFourths/SixtyThreeSixtyFourthsTests.java

```diff
@@ -0,0 +1,218 @@
+  void sixtyThreeSixtyFourthsEcAddTest(
+      long gasLimit, boolean insufficientGasForPrecompileExpected) {
+    final BytecodeCompiler program = BytecodeCompiler.newProgram();
+
+    program.push(4096 - 32).op(OpCode.MLOAD);
+    program
+        .push(0) // returnAtCapacity
+        .push(0) // returnAtOffset
+        .push(0) // callDataSize
+        .push(0) // callDataOffset
+        .push(0) // value
+        .push(ALTBN128_ADD) // address
+        .push(gas) // gas
+        .op(OpCode.CALL);
+    final BytecodeRunner bytecodeRunner = BytecodeRunner.of(program);
+    bytecodeRunner.run(gasLimit);
+
+    // insufficientGasForPrecompileExpected = true  => targetCalleeGas = 63/64 * (252 - 100) = 150
+    // insufficientGasForPrecompileExpected = false => targetCalleeGas = 63/64 * (251 - 100) = 149
+
+    assertNotEquals(
+        OUT_OF_GAS_EXCEPTION,
+        bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+
+    final boolean insufficientGasForPrecompileActual =
+        bytecodeRunner.getHub().oob().operations().stream()
+            .anyMatch(OobOperation::isInsufficientGasForPrecompile);
+
+    assertEquals(insufficientGasForPrecompileExpected, insufficientGasForPrecompileActual);
+  }
+
+  Stream<Arguments> sixtyThreeSixtyFourthsEcAddTestSource() {
+    List<Arguments> arguments = new ArrayList<>();
+    final long targetCalleeGas = PrecompileUtils.getECADDCost();
+    final long gasLimitEnough =
+        getGasLimit(targetCalleeGas, false, true, preCallTargetAddressDoesNotExistProgramGas);
+    final long gasLimitNotEnough =
+        getGasLimit(targetCalleeGas - 1, false, true, preCallTargetAddressDoesNotExistProgramGas);
+    arguments.add(Arguments.of(gasLimitEnough, false));
+    arguments.add(Arguments.of(gasLimitNotEnough, true));
+    return arguments.stream();
+  }
+
+  // Support methods
+  long getGasLimit(
+      long targetCalleeGas,
+      boolean transfersValue,
+      boolean targetAddressExists,
+      long preCallProgramGas) {
+    /* gasLimit = preCallProgramGasCost + gasPreCall
+    /  63/64 * (gasPreCall - gasUpFront) + stipend = targetCalleeGas
+    /  x = gasPreCall - gasUpFront
+    /  k = x / 64 (integer division)
+    /  l = x - 64 * k
+    /  63 * k + l + stipend = targetCalleeGas
+    / find gasLimit going backwards
+    */
+    final long stipend = transfersValue ? GlobalConstants.GAS_CONST_G_CALL_STIPEND : 0;
+    checkArgument(targetCalleeGas >= stipend);
+    final long l = (targetCalleeGas - stipend) % 63;
+    final long k = (targetCalleeGas - stipend - l) / 63;
+    checkArgument(targetCalleeGas == 63 * k + l + stipend);
+    final long gasUpfront = getGasUpfront(transfersValue, targetAddressExists);
+    final long gasPreCall = (targetCalleeGas - stipend) * 64 / 63 + gasUpfront;
+    return preCallProgramGas + gasPreCall; // gasLimit
+  }
+
+  long getGasUpfront(boolean transfersValue, boolean targetAddressExists) {
```

Comment:
We call it `upfrontGasCost`, so I would call it `getUpfrontGasCost` and add a little doc string à la
```java
/**
* {@link #getUpfrontGasCost} computes the upfront gas cost of a non memory expanding {@link OpCode#CALL} to a <b>warm</b> address.
*/
long getUpfrontGasCost(boolean transfersValue, boolean targetAddressExists) { ... }
`
``